### DISCUSSION TITLE
Hide row of aggrid manually?

### DISCUSSION THREAD

**Discussion by tz301** [zauberzeug/nicegui#2843] [Q&A] [answered]
### Question

I have search AG Grid's document and also from stackoverflow:
https://ag-grid.com/javascript-data-grid/filter-external/
https://stackoverflow.com/questions/68821484/hide-or-collapse-a-single-specific-row-in-ag-grid-using-javascript, 

`isExternalFilterPresent/doesExternalFilterPass/onFilterChanged` can be set to implement this feature.

I have tried, but not working. Maybe there are some mistakes.

```py
from nicegui import ui

rows = [
    {'name': 'Alice', 'age': 25},
    {'name': 'Bob', 'age': 30}
]

grid = ui.aggrid({
    'columnDefs': [
        {'field': 'name', 'headerName': 'Name'},
        {'field': 'age', 'headerName': 'Age'},
    ],
    'rowData': rows,
    ':isExternalFilterPresent': '''
        () => {
            return true;
        }
    ''',
    ':doesExternalFilterPass': '''
        (node) => {
            return node.data.age > 28;
        }
    '''
})

grid.run_grid_method('onFilterChanged')  # manually run.

ui.run()
```

**Comment by falkoschindler** [comment, marked as answer]
Hi @tz301,

There seems to be an issue with JavaScript functions starting with a line break. I'll look into it.
Meanwhile, this seems to work:
```py
ui.aggrid({
    'columnDefs': [
        {'field': 'name', 'headerName': 'Name'},
        {'field': 'age', 'headerName': 'Age'},
    ],
    'rowData': [
        {'name': 'Alice', 'age': 25},
        {'name': 'Bob', 'age': 30}
    ],
    ':isExternalFilterPresent': '() => true',
    ':doesExternalFilterPass': '(node) => node.data.age > 28',
})
```
<img width="484" alt="Screenshot 2024-04-07 at 12 28 20" src="https://github.com/zauberzeug/nicegui/assets/5767091/2e9d2162-c954-4c7b-bc78-e525ca016b4f">

**Reply by falkoschindler** [reply]
I created issue https://github.com/zauberzeug/nicegui/issues/2845 to address the problem with leading line breaks.

**Reply by tz301** [reply]
Hi @falkoschindler，

Yeah, this works. Thanks!
